packages/go/dawgs/drivers/pg/driver.go

```diff
@@ -0,0 +1,198 @@
+func (s *Driver) KindMapper() KindMapper {
+	return s.schemaManager
+}
+
+func (s *Driver) SetDefaultGraph(ctx context.Context, graphSchema graph.Graph) error {
+	return s.WriteTransaction(ctx, func(tx graph.Transaction) error {
+		return s.schemaManager.AssertDefaultGraph(tx, graphSchema)
+	})
+}
+
+func (s *Driver) SetBatchWriteSize(size int) {
+}
+
+func (s *Driver) SetWriteFlushSize(size int) {
+}
+
+func (s *Driver) BatchOperation(ctx context.Context, batchDelegate graph.BatchDelegate) error {
+	if cfg, err := renderConfig(readWriteTxOptions, nil); err != nil {
+		return err
+	} else if conn, err := s.pool.Acquire(ctx); err != nil {
+		return err
+	} else {
+		defer conn.Release()
+
+		if batch, err := newBatch(ctx, conn, s.schemaManager, cfg); err != nil {
+			return err
+		} else {
+			defer batch.Close()
+
+			if err := batchDelegate(batch); err != nil {
+				return err
+			}
+
+			return batch.Commit()
+		}
+	}
+}
+
+func (s *Driver) Close(ctx context.Context) error {
+	s.pool.Close()
+	return nil
+}
+
+func renderConfig(pgxOptions pgx.TxOptions, userOptions []graph.TransactionOption) (*Config, error) {
+	graphCfg := graph.TransactionConfig{
+		DriverConfig: &Config{
+			Options:            pgxOptions,
+			QueryExecMode:      pgx.QueryExecModeCacheStatement,
+			QueryResultFormats: pgx.QueryResultFormats{pgx.BinaryFormatCode},
+		},
+	}
+
+	for _, option := range userOptions {
+		option(&graphCfg)
+	}
+
+	if graphCfg.DriverConfig != nil {
+		if pgCfg, typeOK := graphCfg.DriverConfig.(*Config); !typeOK {
+			return nil, fmt.Errorf("invalid driver config type %T", graphCfg.DriverConfig)
+		} else {
+			return pgCfg, nil
+		}
+	}
+
+	return nil, fmt.Errorf("driver config is nil")
+}
+
+const (
+	errStringContextCanceled = "context canceled"
+)
+
+func (s *Driver) ReadTransaction(ctx context.Context, txDelegate graph.TransactionDelegate, options ...graph.TransactionOption) error {
+	if cfg, err := renderConfig(readOnlyTxOptions, options); err != nil {
+		return err
+	} else if conn, err := s.pool.Acquire(ctx); err != nil {
+		return err
+	} else {
+		defer conn.Release()
+
+		return txDelegate(&transaction{
+			schemaManager:   s.schemaManager,
+			queryExecMode:   cfg.QueryExecMode,
+			ctx:             ctx,
+			conn:            conn,
+			targetSchemaSet: false,
+		})
+	}
+}
+
+func (s *Driver) WriteTransaction(ctx context.Context, txDelegate graph.TransactionDelegate, options ...graph.TransactionOption) error {
+	if cfg, err := renderConfig(readWriteTxOptions, options); err != nil {
+		return err
+	} else if conn, err := s.pool.Acquire(ctx); err != nil {
+		return err
+	} else {
+		defer conn.Release()
+
+		if tx, err := newTransaction(ctx, conn, s.schemaManager, cfg); err != nil {
+			return err
+		} else {
+			defer tx.Close()
+
+			if err := txDelegate(tx); err != nil {
+				return err
+			}
+
+			return tx.Commit()
+		}
+	}
+}
+
+func (s *Driver) FetchSchema(ctx context.Context) (graph.Schema, error) {
+	return graph.Schema{}, nil
+}
```

Comment:
Is this meant to return an empty `graph.Schema`? I think we'd want this API to return the schema that had been previously asserted.

Reply:
This is going to have to wait until I have time to formally flesh this function out. Currently, most of the details of the schema are managed by the `SchemaManager` type. This is a future compatibility feature.

Reply:
TODO with note added